core/src/io/anuke/mindustry/entities/type/TileEntity.java

```diff
@@ -307,14 +307,17 @@ public void update(){
         }
 
         Block previous = block;
-        block.update(tile);
-        if(block == previous && cons != null){
+        if(cons != null){
             cons.update();
         }
 
-        if(block == previous && power != null){
-            power.graph.update();
+        float amountOfPowerUsed = 0.0f;
+        
+        if(power != null){
+            amountOfPowerUsed = power.graph.update();
         }
+        
+    	block.update(tile, amountOfPowerUsed);
```

Comment:
Hold on. If you have, say, 10 generators giving just enough power to supply a block with power, then one of the generators will have positive power used, and the other 9 will have 0, as the graph just updated, making them stop updating next frame. This will lead to all sorts of issues.